src/internal/connector/data_collections.go

```diff
@@ -247,58 +261,42 @@ func (gc *GraphConnector) OneDriveDataCollections(
 func (gc *GraphConnector) createSharePointCollections(
 	ctx context.Context,
 	scope selectors.SharePointScope,
-) ([]*sharepoint.Collection, error) {
+) ([]data.Collection, error) {
 	var (
-		errs  *multierror.Error
-		sites = scope.Get(selectors.SharePointSite)
-		colls = make([]*sharepoint.Collection, 0)
+		errs        *multierror.Error
+		sites       = scope.Get(selectors.SharePointSite)
+		category    = scope.Category().PathType()
+		collections = make([]data.Collection, 0)
 	)
 
 	// Create collection of ExchangeDataCollection
 	for _, site := range sites {
-		collections := make(map[string]*sharepoint.Collection)
-
-		qp := graph.QueryParams{
-			Category:      scope.Category().PathType(),
-			ResourceOwner: site,
-			FailFast:      gc.failFast,
-			Credentials:   gc.credentials,
-		}
-
-		foldersComplete, closer := observe.MessageWithCompletion(fmt.Sprintf("∙ %s - %s:", qp.Category, site))
+		foldersComplete, closer := observe.MessageWithCompletion(fmt.Sprintf("∙ %s - %s:", category, site))
 		defer closer()
 		defer close(foldersComplete)
 
-		// resolver, err := exchange.PopulateExchangeContainerResolver(
-		// 	ctx,
-		// 	qp,
-		// 	qp.Scope.Category().PathType(),
-		// )
-		// if err != nil {
-		// 	return nil, errors.Wrap(err, "getting folder cache")
-		// }
-
-		// err = sharepoint.FilterContainersAndFillCollections(
-		// 	ctx,
-		// 	qp,
-		// 	collections,
-		// 	gc.UpdateStatus,
-		// 	resolver)
-
-		// if err != nil {
-		// 	return nil, errors.Wrap(err, "filling collections")
-		// }
-
-		foldersComplete <- struct{}{}
-
-		for _, collection := range collections {
-			gc.incrementAwaitingMessages()
+		switch category {
+		case path.FilesCategory: // TODO: better category for drives
```

Comment:
can you give a sketch of what you'd like to use as the category instead? Also rather unfortunately, as the category is persisted as part of the reporef of an item changing it will be a breaking change

Reply:
Ah, I'm using terminology loosely here.  `drives` in this case mean sharePoint drives, specifically.  The todo is not meant to reflect upon oneDrive terminology/consts.

The categorical division in sharepoint is currently Libraries and Sites.  Libraries contain the drive-based data, Sites contain the webItem data.  Very likely, the todo will be to swap `FilesCategory` for `LibrariesCategory`.